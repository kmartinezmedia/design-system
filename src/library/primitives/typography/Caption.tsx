import React, { FunctionComponent } from 'react';
import { StyleSheet } from 'react-native';
import { Text, TextProps } from './Text';
import {
  fontFamily,
  fontSize,
  lineHeight,
  fontWeight,
} from '@designSystem/theme';

export const Caption: FunctionComponent<TextProps> = ({
  children,
  color = 'muted',
  spacing = { bottom: 1 },
  ...typeProps
}) => (
  <Text {...typeProps} spacing={spacing} color={color} style={styles.Caption}>
    {children}
  </Text>
);

const styles = StyleSheet.create({
  Caption: {
    fontFamily: fontFamily.regular,
    fontWeight: fontWeight.regular,
    fontSize: fontSize.caption,
    lineHeight: lineHeight.caption,
  },
});
