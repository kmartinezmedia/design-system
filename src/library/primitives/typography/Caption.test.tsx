import React from 'react';
import 'react-native';
import { create, act, ReactTestRenderer } from 'react-test-renderer';
import { Caption } from './Caption';

describe('Caption', () => {
  it('default', () => {
    let component: ReactTestRenderer;
    act(() => {
      component = create(<Caption>Hey, lil 🐛</Caption>);
    });
    expect(component!.toJSON()).toMatchSnapshot();
  });
});
