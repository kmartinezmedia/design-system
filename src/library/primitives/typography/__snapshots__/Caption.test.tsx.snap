// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Caption default 1`] = `
<Text
  style={
    Array [
      Object {
        "fontFamily": "Graphik-Regular",
        "fontSize": 13,
        "fontWeight": "400",
        "lineHeight": 16,
      },
      Object {
        "paddingBottom": 4,
      },
      Object {
        "color": "#587088",
        "lineHeight": 16,
        "overflow": "visible",
        "textAlign": undefined,
      },
      undefined,
    ]
  }
>
  Hey, lil 🐛
</Text>
`;
